.github/workflows/code-coverage.yml

```diff
@@ -39,6 +39,9 @@ jobs:
     - name: dotnet test
       run: dotnet test --collect:"Code Coverage" --results-directory:"TestResults" --configuration Release --no-build -- RunConfiguration.DisableAppDomain=true
 
+    - name: Install coverage tool
+      run: dotnet tool install -g dotnet-coverage
```

Comment:
Funny I have been messing with this too over on https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1347. I think we can go deeper with the changes:

* Remove `.\build\process-codecoverage.ps1`
* Remove `reportgenerator` completely

It seems `dotnet-coverage merge` is able to do all that work. Here is what I have so far on contrib:

```yaml
    - name: Setup dotnet
      uses: actions/setup-dotnet@v3

    - name: Install dependencies
      run: dotnet restore

    - name: dotnet build
      run: dotnet build --configuration Release --no-restore

    - name: dotnet test
      run: dotnet test --collect:"Code Coverage" --results-directory:"TestResults" --configuration Release --no-restore --no-build -- RunConfiguration.DisableAppDomain=true

    - name: Install coverage tool
      run: dotnet tool install -g dotnet-coverage

    - name: Merging test results
      run: dotnet-coverage merge -r -f cobertura -o ./TestResults/Cobertura.xml ./TestResults/*.coverage

    - uses: codecov/codecov-action@v3.1.4
      env:
        OS: ${{ matrix.os }}
      with:
        file: TestResults/Cobertura.xml
        env_vars: OS
        flags: unittests
        name: Code Coverage for ${{ matrix.os }}
```

PS: Note the addition of `env:` in the upload. I don't think the current `env_vars` has ever worked without that.

Reply:
To be honest I didn't checked your PR in contrib. It was in draft state :)

Suggestions Aoplied.